background.js

```diff
@@ -1,36 +1,42 @@
-chrome.browserAction.setBadgeText({text: 'OFF'});
-chrome.browserAction.setBadgeBackgroundColor({color: '#4688F1'});
+chrome.browserAction.setBadgeText({ text: "OFF" });
+chrome.browserAction.setBadgeBackgroundColor({ color: "#4688F1" });
 let reloads_holder = {};
+let count = 0;
+function callback(element) {
+  if (chrome.runtime.lastError) {
+    console.log(chrome.runtime.lastError.message);
+    clearInterval(reloads_holder[element]);
+  }
+}
 
 chrome.storage.onChanged.addListener((whatchanged, area) => {
-    if(area==="local"){
-        let keys = Object.keys(whatchanged);
-        keys.forEach(element => {
-            console.log(whatchanged[element]);
-            if(whatchanged[element].newValue){
-                reloads_holder[element] = setInterval(()=>{
-                    chrome.tabs.reload(parseInt(element));
-                }, 10000);
-            }
-            else {
-                clearInterval(reloads_holder[element]);
-            }
-        });
+  console.log(area);
+  if (area === "local") {
+    let keys = Object.keys(whatchanged);
+    keys.forEach((element) => {
+      console.log(element);
+      console.log(whatchanged[element].newValue);
 
-        
-    }
+      if (whatchanged[element].newValue) {
+        reloads_holder[element] = setInterval(() => {
+          chrome.tabs.reload(parseInt(element), () => callback(element));
```

Comment:
This works fine!
Can it be any better if we use chrome.tabs.onRemoved? https://developer.chrome.com/extensions/tabs#event-onRemoved
Sure this works and I do not have a problem with this approach specifically, but this is more of catch thing instead of a voluntary action of removing the tab from the watchList.

I would like to know your thoughts on this.